src/hub.ts

```diff
@@ -175,8 +179,8 @@ export class Hub extends EventEmitter {
      */
     public unsubscribe (port: string) {
         return new Promise((resolve, reject) => {
-            const mode = this._getModeForDeviceType(this._portLookup(port).type);
-            this._deactivatePortDevice(this._portLookup(port).value, this._portLookup(port).type, mode, 0x00, () => {
+            const { type, value, mode } = this._portLookup(port);
+            this._deactivatePortDevice(value, type, mode || this._getModeForDeviceType(type), 0x00, () => {
```

Comment:
This will switch the port mode to the default for type (`_getModeForDeviceType`) if the `mode` is 0 since 0 is falsy.
Technically it should be `mode === null` but this could only happen when you call `unsubscribe` before `subscribe`, so not sure if that should be handled.

Reply:
I am not sure to understand how subscriptions work: I don't get why mode is needed to unsubscribe...